### PR TITLE
[WIP] Implementing partner stats api for archetypes

### PR DESCRIPTION
**Pull request by azeier**
This is build on top of dev/partner-auth.

Part of #864.

### CODE REVIEW

File: hsreplaynet/static/scripts/src/DataManager.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default class DataManager {
 			);
 		}, "");
 
-		return url + query;
+		return "https://hsreplay.net" + url + query;
```
Comment (jleclanche) [inline hsreplaynet/static/scripts/src/DataManager.ts]: Careful what you land

File: hsreplaynet/admin/management/commands/grant_partner_access.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		if not options["noinput"]:
+			if input(
+				'Create partner access token for "%s" using application "%s"? [y/N] ' % (
+					str(user), str(application)
+				)
+			).lower() != "y":
+				raise CommandError("Aborted")
+
+		with transaction.atomic():
+			user.user_permissions.add(permission)
```
Comment (jleclanche) [inline hsreplaynet/admin/management/commands/grant_partner_access.py]: This needs to be updated with @beheh's new plan to use Feature tags.

File: hsreplaynet/api/urls.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 from rest_framework.routers import DefaultRouter
 
 from hsreplaynet.analytics.urls import api_urlpatterns as analytics_urlpatterns
+from hsreplaynet.api.partner.views import ArchetypesView
```
Comment (beheh) [inline hsreplaynet/api/urls.py]: Remove this.

File: hsreplaynet/api/urls.py [hunk 4]
```diff
@@ -44,7 +45,7 @@
 	url(r"^v1/analytics/personal/$", views.analytics.PersonalAnalyticsQueryView.as_view()),
 	url(r"^v1/features/(?P<name>[\w-]+)/$", SetFeatureView.as_view()),
 	url(r"^v1/live/", include("hsreplaynet.api.live.urls")),
-	url(r"^api-auth/", include("rest_framework.urls", namespace="rest_framework")),
```
Comment (beheh) [inline hsreplaynet/api/urls.py]: Don't think this change was meant here.

File: hsreplaynet/api/partner/views.py [hunk 5]
```diff
@@ -17,3 +27,67 @@ def get(self, request, format=None):
...
+class ArchetypesView(ListAPIView):
+	authentication_classes = (OAuth2Authentication, )
+	permission_classes = (TokenHasScope, PartnerPermission)
+	required_scopes = ["stats.partner:read"]
+	pagination_class = None
+	serializer_class = ArchetypesSerializer
+
+	supported_game_types = ["RANKED_STANDARD"]
+
+	class QueryDataNotAvailableException(Exception):
```
Comment (beheh) [inline hsreplaynet/api/partner/views.py]: Should probably move this to a place where all the views can use it.

File: tests/test_partner_api.py [hunk 6]
```diff
@@ -42,3 +48,202 @@ def test_partner_api_example(partner_token, client):
...
+	)
+	assert response.status_code == status.HTTP_200_OK
+	assert response.data
+
+
+def test_archetypes_view_not_authorized(client):
+	response = client.get("/api/v1/partner/archetypes/")
+	assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+# TODO: Add tests 429
```
Comment (beheh) [inline tests/test_partner_api.py]: Don't forget this.

Comment (azeier) [reply tests/test_partner_api.py]: There will be no throttling on this endpoint specifically, so this no longer makes sense.

File: hsreplaynet/static/scripts/src/DataManager.ts (hunk 1 above)
Comment (azeier) [reply hsreplaynet/static/scripts/src/DataManager.ts]: thanks

**Pull request merged**